src/Module/Api/Mastodon/Timelines/Home.php

```diff
@@ -96,11 +87,19 @@ protected function rawContent(array $request = [])
 
 		$statuses = [];
 		while ($item = Post::fetch($items)) {
-			self::setBoundaries($item['uri-id']);
 			try {
-				$statuses[] = DI::mstdnStatus()->createFromUriId($item['uri-id'], $uid, $display_quotes);
-			} catch (\Exception $exception) {
-				Logger::info('Post not fetchable', ['uri-id' => $item['uri-id'], 'uid' => $uid, 'exception' => $exception]);
+				$status =  DI::mstdnStatus()->createFromUriId($item['uri-id'], $uid, $display_quotes);
+				switch ($request['friendica_order']) {
+					case TimelineOrderByTypes::CREATED:
+						self::setBoundaries($status->createdAtTimestamp());
```

Comment:
Why not set the boundary as a full date object? This prevents you from having to extract the timestamp and then creating another object from the timestamp.

Reply:
Can one do min/max calculations on DateTime objects in PHP? If so that would be even better than the current conversion stuff.

Reply:
I think I got it to just work. The only snafu is that the type of the parameter of the min function has to be unspecified. I tried the compound type thing but apparently that is only a PHP 8 and above thing. But I added a hint to the documentation.